modules/db_instance/main.tf

```diff
@@ -1,6 +1,24 @@
 ##############
 # DB instance
 ##############
+
+locals {
+  create_enhanced_role = "${var.monitoring_role_autocreate != "" && var.monitoring_role_arn == ""}"
```

Comment:
Avoid such rules (and `locals` in general unless it is possible to use specific module versions in Terraform Registry). Use `count = "${var.create_monitoring_role}"` as the main switch.

Reply:
I wanted to avoid the case of misconfiguration when a user provides ARN and also configure auto-creation, which could be skipped, but sure I will remove it.